src/query/models/tag.go

```diff
@@ -218,35 +238,75 @@ func (t Tags) IDWithKeys(includeKeys ...string) uint64 {
 
 // TagsWithKeys returns only the tags which have the given keys
 func (t Tags) TagsWithKeys(includeKeys []string) Tags {
-	includeTags := make(Tags, len(includeKeys))
-	for _, k := range includeKeys {
-		if v, ok := t[k]; ok {
-			includeTags[k] = v
+	return t.tagsSubSet(includeKeys, true)
+
+}
+
+// WithoutName copies the tags excluding the name tag
+func (t Tags) WithoutName() Tags {
+	return t.TagsWithoutKeys([]string{MetricName})
+}
+
+// Get returns the value for the tag with the given name.
+func (t Tags) Get(key string) (string, bool) {
+	for _, tag := range t {
+		if tag.Name == key {
+			return tag.Value, true
 		}
 	}
-
-	return includeTags
+	return "", false
 }
 
-func (t Tags) sortKeys() ([]string, int) {
-	length := 0
-	keys := make([]string, 0, len(t))
-	for k := range t {
-		length += len(k) + len(t[k]) + 2
-		keys = append(keys, k)
+// FromMap returns new sorted tags from the given map.
+func FromMap(m map[string]string) Tags {
+	l := make([]Tag, 0, len(m))
+	for k, v := range m {
+		l = append(l, Tag{Name: k, Value: v})
 	}
-	sort.Strings(keys)
-	return keys, length
+	return New(l)
 }
 
-// WithoutName copies the tags excluding the name tag
-func (t Tags) WithoutName() Tags {
-	tags := make(Tags, len(t))
-	for k, v := range t {
-		if k == MetricName {
-			continue
-		}
-		tags[k] = v
+// MultiTagsFromMaps returns a slice of tags from a slice of maps
+func MultiTagsFromMaps(tagMaps []map[string]string) []Tags {
+	tags := make([]Tags, len(tagMaps))
+	for i, m := range tagMaps {
+		tags[i] = FromMap(m)
 	}
+
 	return tags
 }
+
+// TagMap returns a tag map of the tags.
+func (t Tags) TagMap() map[string]Tag {
+	m := make(map[string]Tag, len(t))
+	for _, tag := range t {
+		m[tag.Name] = tag
+	}
+
+	return m
+}
+
+// StringMap returns a string map of the tags.
```

Comment:
might still be good to have

Reply:
I'd delete it until necessary, otherwise might have a bunch of unnecessary functions hanging around for all time because we don't realize they're not needed